mobile/mobile.go

```diff
@@ -0,0 +1,505 @@
+package mobile
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"regexp"
+	"runtime/debug"
+	"strings"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/golang/protobuf/proto"
+	"github.com/lightninglabs/faraday/frdrpc"
+	"github.com/lightninglabs/lightning-node-connect/core"
+	"github.com/lightninglabs/lightning-node-connect/mailbox"
+	"github.com/lightninglabs/loop/looprpc"
+	"github.com/lightninglabs/pool/poolrpc"
+	"github.com/lightningnetwork/lnd/build"
+	"github.com/lightningnetwork/lnd/keychain"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/autopilotrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/chainrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/invoicesrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/routerrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/signrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/verrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/walletrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/watchtowerrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/wtclientrpc"
+	"github.com/lightningnetwork/lnd/signal"
+	"google.golang.org/grpc"
+	"gopkg.in/macaroon-bakery.v2/bakery"
+	"gopkg.in/macaroon-bakery.v2/bakery/checkers"
+	"gopkg.in/macaroon.v2"
+)
+
+type mobileClient struct {
+	lndConn *grpc.ClientConn
+
+	statusChecker func() mailbox.ClientStatus
+
+	mac *macaroon.Macaroon
+
+	registry map[string]func(context.Context, *grpc.ClientConn,
+		string, func(string, error))
+}
+
+type stubPackageRegistration func(map[string]func(context.Context,
+	*grpc.ClientConn, string, func(string, error)))
+
+type NativeCallback interface {
+	SendResult(json string)
+}
+
+var (
+	registrations = []stubPackageRegistration{
+		lnrpc.RegisterLightningJSONCallbacks,
+		lnrpc.RegisterStateJSONCallbacks,
+		autopilotrpc.RegisterAutopilotJSONCallbacks,
+		chainrpc.RegisterChainNotifierJSONCallbacks,
+		invoicesrpc.RegisterInvoicesJSONCallbacks,
+		routerrpc.RegisterRouterJSONCallbacks,
+		signrpc.RegisterSignerJSONCallbacks,
+		verrpc.RegisterVersionerJSONCallbacks,
+		walletrpc.RegisterWalletKitJSONCallbacks,
+		watchtowerrpc.RegisterWatchtowerJSONCallbacks,
+		wtclientrpc.RegisterWatchtowerClientJSONCallbacks,
+		looprpc.RegisterSwapClientJSONCallbacks,
+		poolrpc.RegisterTraderJSONCallbacks,
+		frdrpc.RegisterFaradayServerJSONCallbacks,
+	}
+
+	perms = core.GetAllMethodPermissions()
+
+	jsonCBRegex = regexp.MustCompile(`(\w+)\.(\w+)\.(\w+)`)
+
+	m *mobileClient
+
+	registry = make(map[string]func(context.Context,
+		*grpc.ClientConn, string, func(string, error)))
+
+	localPrivCreateCallback  NativeCallback
+	remoteKeyReceiveCallback NativeCallback
+	authDataCallback         NativeCallback
```

Comment:
must we use global variables here? will we not ever have a situation where we would like to create more than 1 LNC connection at a time? 
If you take a look at the `wasmClient`, it has a `cfg` variable in which we store callback options. &we make use of namespaces so that we can have a connection for different namespaces

Reply:
I wanted to manage this at the JS level but it appears that React Native Native Modules are singletons. I've added support for multiple connections at https://github.com/lightninglabs/lightning-node-connect/pull/52/commits/7b48fb7bfc35f12c90a37cc9858074030e711f79, please let me know what you think.

The use of namespace params for all exposed functions instead of class functions is required because of restrictions in gomobile